src/main/scala/it/unibo/authsim/library/user/builder/CostumUserBuilder.scala

```diff
@@ -0,0 +1,43 @@
+package it.unibo.authsim.library.user.builder
+import it.unibo.authsim.library.dsl.policy.builders.PolicyBuilder
+import it.unibo.authsim.library.dsl.policy.checkers.StringPolicyChecker
+import it.unibo.authsim.library.dsl.policy.model.StringPolicies.{CredentialPolicy, PasswordPolicy, UserIDPolicy}
+import it.unibo.authsim.library.user.model.User
+import it.unibo.authsim.library.user.builder.util.Util.generateRandomString
+
+//TODO controlla se con i metodi del builder si può parlare di setter
+/**
+ * Class that represent a costum builder for a user
+ */
+class UserCostumBuilder extends UserBuilder[User]:
+  /**
+   * Setter for the username of the user
+   * 
+   * @param userName    userName to use for the generation of the new user
+   * @return            a UserCostumBuilder where the username field is setted with the provided value
+   */
+  def withName(userName:String):this.type=
+    this._userName = userName
```

Comment:
This underscore is not needed as `this.userName` does not conflict with `userName`.

The same goes with `withPassword` method